Index: drscheme-371/collects/web-server/default-web-root/configuration-table.ss
===================================================================
--- drscheme-371.orig/collects/web-server/default-web-root/configuration-table.ss	2007-11-21 22:59:02.000000000 +0000
+++ drscheme-371/collects/web-server/default-web-root/configuration-table.ss	2007-11-21 22:59:11.000000000 +0000
@@ -1,4 +1,4 @@
-((port 80)
+((port 8000)
  (max-waiting 40)
  (initial-connection-timeout 30)
  (default-host-table
