Index: drscheme-360/src/configure
===================================================================
--- drscheme-360.orig/src/configure	2006-08-28 05:42:12.000000000 -0400
+++ drscheme-360/src/configure	2007-04-02 20:56:37.000000000 -0400
@@ -5369,7 +5369,7 @@
        fi
        ;;
       alpha)
-       EXTRA_GMP_OBJ="gmp_alpha_gcc.o"
+       EXTRA_GMP_OBJ="gmp_alpha_gcc.lo"
        ;;
       *)
        ;;
Index: drscheme-360/src/mzscheme/src/Makefile.in
===================================================================
--- drscheme-360.orig/src/mzscheme/src/Makefile.in	2006-10-13 18:03:29.000000000 -0400
+++ drscheme-360/src/mzscheme/src/Makefile.in	2007-04-02 20:56:37.000000000 -0400
@@ -6,6 +6,7 @@
 builddir = @builddir@
 
 CC = @CC@
+AS = @CC@
 PERL = @PERL@
 
 XMAKE = $(MAKE) $(EXPORT)
@@ -221,7 +222,7 @@
 	$(CC) $(CFLAGS) -c $(srcdir)/vector.c -o vector.@LTO@
 
 gmp_alpha_gcc.@LTO@: $(srcdir)/gmp/gmp_alpha_gcc.s
-	$(AS) -o gmp_alpha_gcc.@LTO@ $(srcdir)/gmp/gmp_alpha_gcc.s
+	$(AS) -o gmp_alpha_gcc.@LTO@ -c $(srcdir)/gmp/gmp_alpha_gcc.s
 
 SCONFIG =   $(srcdir)/../sconfig.h $(srcdir)/../uconfig.h
 
