Index: drscheme-371/src/configure
===================================================================
--- drscheme-371.orig/src/configure	2007-07-23 19:02:09.000000000 +0000
+++ drscheme-371/src/configure	2007-11-21 23:08:10.000000000 +0000
@@ -5398,7 +5398,7 @@
        fi
        ;;
       alpha)
-       EXTRA_GMP_OBJ="gmp_alpha_gcc.o"
+       EXTRA_GMP_OBJ="gmp_alpha_gcc.lo"
        ;;
       *)
        ;;
Index: drscheme-371/src/mzscheme/src/Makefile.in
===================================================================
--- drscheme-371.orig/src/mzscheme/src/Makefile.in	2007-06-19 21:31:22.000000000 +0000
+++ drscheme-371/src/mzscheme/src/Makefile.in	2007-11-21 23:08:10.000000000 +0000
@@ -6,6 +6,7 @@
 builddir = @builddir@
 
 CC = @CC@
+AS = @CC@
 PERL = @PERL@
 
 MZSRC = $(srcdir)
@@ -222,7 +223,7 @@
 	$(CC) $(CFLAGS) -c $(srcdir)/vector.c -o vector.@LTO@
 
 gmp_alpha_gcc.@LTO@: $(srcdir)/gmp/gmp_alpha_gcc.s
-	$(AS) -o gmp_alpha_gcc.@LTO@ $(srcdir)/gmp/gmp_alpha_gcc.s
+	$(AS) -o gmp_alpha_gcc.@LTO@ -c $(srcdir)/gmp/gmp_alpha_gcc.s
 
 SCONFIG =   $(srcdir)/../sconfig.h $(srcdir)/../uconfig.h
 
