Index: drscheme-4.0.1/src/configure
===================================================================
--- drscheme-4.0.1.orig/src/configure
+++ drscheme-4.0.1/src/configure
@@ -5798,7 +5798,7 @@ case $OS in
        fi
        ;;
       alpha)
-       EXTRA_GMP_OBJ="gmp_alpha_gcc.o"
+       EXTRA_GMP_OBJ="gmp_alpha_gcc.lo"
        ;;
       *)
        ;;
Index: drscheme-4.0.1/src/mzscheme/src/Makefile.in
===================================================================
--- drscheme-4.0.1.orig/src/mzscheme/src/Makefile.in
+++ drscheme-4.0.1/src/mzscheme/src/Makefile.in
@@ -6,6 +6,7 @@ srcdir = @srcdir@
 builddir = @builddir@
 
 CC = @CC@
+AS = @CC@
 PERL = @PERL@
 
 MZSRC = $(srcdir)
@@ -222,7 +223,7 @@ vector.@LTO@: $(srcdir)/vector.c
 	$(CC) $(CFLAGS) -c $(srcdir)/vector.c -o vector.@LTO@
 
 gmp_alpha_gcc.@LTO@: $(srcdir)/gmp/gmp_alpha_gcc.s
-	$(AS) -o gmp_alpha_gcc.@LTO@ $(srcdir)/gmp/gmp_alpha_gcc.s
+	$(AS) -o gmp_alpha_gcc.@LTO@ -c $(srcdir)/gmp/gmp_alpha_gcc.s
 
 SCONFIG =   $(srcdir)/../sconfig.h $(srcdir)/../uconfig.h
 
