Index: drscheme-4.0.1/src/mred/wxs/Makefile.in
===================================================================
--- drscheme-4.0.1.orig/src/mred/wxs/Makefile.in
+++ drscheme-4.0.1/src/mred/wxs/Makefile.in
@@ -29,8 +29,8 @@ WXINC_wx_mac = -I$(PLTSRCDIR)/wxmac/incl
 WXINC = $(WXINC_@WXVARIANT@)
 NOGCINC = $(WXINC) -I$(PLTSRCDIR)/mred/wxme/ -I$(PLTBUILDDIR)/mzscheme/ -I$(PLTSRCDIR)/mzscheme/include/
 INC = -I$(PLTSRCDIR)/mzscheme/$(GCDIRECTORY) $(NOGCINC)
-CXXFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -D$(WXVARIANT) @COMPFLAGS@ @PROFFLAGS@ @CXXFLAGS@ @PREFLAGS@
-CFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -D$(WXVARIANT) @COMPFLAGS@ @PROFFLAGS@ @CFLAGS@ @PREFLAGS@
+CXXFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -I/usr/include/freetype2 -D$(WXVARIANT) @COMPFLAGS@ @PROFFLAGS@ @CXXFLAGS@ @PREFLAGS@
+CFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -I/usr/include/freetype2 -D$(WXVARIANT) @COMPFLAGS@ @PROFFLAGS@ @CFLAGS@ @PREFLAGS@
 LDFLAGS = $(XLIB)
 ###########################
 
Index: drscheme-4.0.1/src/wxxt/src/Makefile.in
===================================================================
--- drscheme-4.0.1.orig/src/wxxt/src/Makefile.in
+++ drscheme-4.0.1/src/wxxt/src/Makefile.in
@@ -25,8 +25,8 @@ WBUILD = @WBUILD@
 WXINC = $(PLTSRCDIR)/wxxt/src/AIAI-include -I$(PLTSRCDIR)/wxxt/src
 NOGCINC = -I$(WXINC) -I$(PLTSRCDIR)/mred/wxme/ -I$(PLTBUILDDIR)/mzscheme/ -I$(PLTSRCDIR)/mzscheme/include/
 INC = -I$(PLTSRCDIR)/mzscheme/$(GCDIRECTORY) $(NOGCINC)
-CXXFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -Dwx_xt @COMPFLAGS@ @PROFFLAGS@ @CXXFLAGS@ @PREFLAGS@
-CFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -Dwx_xt @COMPFLAGS@ @PROFFLAGS@ @CFLAGS@ @PREFLAGS@
+CXXFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -I/usr/include/freetype2 -Dwx_xt @COMPFLAGS@ @PROFFLAGS@ @CXXFLAGS@ @PREFLAGS@
+CFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -I/usr/include/freetype2 -Dwx_xt @COMPFLAGS@ @PROFFLAGS@ @CFLAGS@ @PREFLAGS@
 LDFLAGS = $(XLIB)
 ###########################
 
