Index: drscheme-371/src/mred/wxs/Makefile.in
===================================================================
--- drscheme-371.orig/src/mred/wxs/Makefile.in	2007-02-06 03:39:37.000000000 +0000
+++ drscheme-371/src/mred/wxs/Makefile.in	2007-11-21 23:09:10.000000000 +0000
@@ -29,8 +29,8 @@
 WXINC = $(WXINC_@WXVARIANT@)
 NOGCINC = $(WXINC) -I$(PLTSRCDIR)/mred/wxme/ -I$(PLTBUILDDIR)/mzscheme/ -I$(PLTSRCDIR)/mzscheme/include/
 INC = -I$(PLTSRCDIR)/mzscheme/$(GCDIRECTORY) $(NOGCINC)
-CXXFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -D$(WXVARIANT) @COMPFLAGS@ @PROFFLAGS@ @CXXFLAGS@ @PREFLAGS@
-CFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -D$(WXVARIANT) @COMPFLAGS@ @PROFFLAGS@ @CFLAGS@ @PREFLAGS@
+CXXFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -I/usr/include/freetype2 -D$(WXVARIANT) @COMPFLAGS@ @PROFFLAGS@ @CXXFLAGS@ @PREFLAGS@
+CFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -I/usr/include/freetype2 -D$(WXVARIANT) @COMPFLAGS@ @PROFFLAGS@ @CFLAGS@ @PREFLAGS@
 LDFLAGS = $(XLIB)
 ###########################
 
Index: drscheme-371/src/wxxt/src/Makefile.in
===================================================================
--- drscheme-371.orig/src/wxxt/src/Makefile.in	2007-01-12 07:09:56.000000000 +0000
+++ drscheme-371/src/wxxt/src/Makefile.in	2007-11-21 23:09:10.000000000 +0000
@@ -25,8 +25,8 @@
 WXINC = $(PLTSRCDIR)/wxxt/src/AIAI-include -I$(PLTSRCDIR)/wxxt/src
 NOGCINC = -I$(WXINC) -I$(PLTSRCDIR)/mred/wxme/ -I$(PLTBUILDDIR)/mzscheme/ -I$(PLTSRCDIR)/mzscheme/include/
 INC = -I$(PLTSRCDIR)/mzscheme/$(GCDIRECTORY) $(NOGCINC)
-CXXFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -Dwx_xt @COMPFLAGS@ @PROFFLAGS@ @CXXFLAGS@ @PREFLAGS@
-CFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -Dwx_xt @COMPFLAGS@ @PROFFLAGS@ @CFLAGS@ @PREFLAGS@
+CXXFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -I/usr/include/freetype2 -Dwx_xt @COMPFLAGS@ @PROFFLAGS@ @CXXFLAGS@ @PREFLAGS@
+CFLAGS = $(INC) $(XINCLUDE) $(OPTIONS) -I/usr/include/freetype2 -Dwx_xt @COMPFLAGS@ @PROFFLAGS@ @CFLAGS@ @PREFLAGS@
 LDFLAGS = $(XLIB)
 ###########################
 
