--- branches/v4.0.2/collects/profj/libs/java/lang/Object-composite.ss	2008/07/03 20:07:53	10585
+++ branches/v4.0.2/collects/profj/libs/java/lang/Object-composite.ss	2008/07/03 20:55:50	10586
@@ -2,7 +2,7 @@
   
   (require scheme/class
            (prefix-in c: scheme/contract)
-           errortrace
+           errortrace/errortrace-lib
            
            profj/libs/java/lang/Comparable
            profj/libs/java/io/Serializable)
