Index: drscheme-360/src/configure
===================================================================
--- drscheme-360.orig/src/configure	2007-04-02 21:29:29.000000000 -0400
+++ drscheme-360/src/configure	2007-04-02 21:29:54.000000000 -0400
@@ -8335,7 +8335,7 @@
        return grows_down_p(n - 1, cmp);
    }
    int main() {
-      return grows_down_p(0, 0);
+      return grows_down_p(0, (void*)0);
    }
 _ACEOF
 rm -f conftest$ac_exeext
Index: drscheme-360/src/mzscheme/src/setjmpup.c
===================================================================
--- drscheme-360.orig/src/mzscheme/src/setjmpup.c	2007-04-02 21:24:34.000000000 -0400
+++ drscheme-360/src/mzscheme/src/setjmpup.c	2007-04-02 21:33:34.000000000 -0400
@@ -360,6 +360,7 @@
       top_delta = (((unsigned long)c->cont->buf.stack_from
 		    + c->cont->buf.stack_size)
 		   - (unsigned long)c->stack_from);
+	  top_delta = -top_delta;
 #else
       bottom_delta = ((unsigned long)c->stack_from 
 		      + c->stack_size
