--- drscheme-301.orig/collects/slibinit/init.ss
+++ drscheme-301/collects/slibinit/init.ss
@@ -71,8 +71,10 @@
 	  ;; Use this path if your scheme does not support GETENV
 	  (with-handlers ([void
 			   (lambda (x)
-			     (error 'slib-init
-				    "can't find SCHEME_LIBRARY_PATH environment variable or \"slib\" collection"))])
+			     ;;(error 'slib-init
+				 ;;   "can't find SCHEME_LIBRARY_PATH environment variable or \"slib\" collection"))])
+				 ;; Debian SLIB path
+				 "/usr/share/slib")])
 	    (path->string (collection-path "slib"))))))
     (lambda () library-path)))

