diff -Nur drscheme-350/src/Makefile.in drscheme-350.new/src/Makefile.in
--- drscheme-350/src/Makefile.in	2006-06-16 03:28:12.000000000 +0000
+++ drscheme-350.new/src/Makefile.in	2006-06-22 17:18:08.000000000 +0000
@@ -49,7 +49,7 @@
 mredinstall-stub: @MAKE_MREDINSTALL@
 
 setup-plt:
-	$(prefix)/bin/mzscheme -mvqM setup
+	LD_LIBRARY_PATH="$(prefix)/lib" $(prefix)/bin/mzscheme -mvqM setup
 
 plain-install-3m:
 	$(MAKE) install-normal
